File: src/main/java/com/mohe/spring/entity/Bookmark.java
package com.mohe.spring.entity;

import jakarta.persistence.*;
import java.time.OffsetDateTime;

@Entity
@Table(name = "bookmarks", 
       uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "place_id"}))
public class Bookmark {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id = 0L;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "place_id", nullable = false)
    private Place place;
    
    @Column(name = "created_at")
    private OffsetDateTime createdAt = OffsetDateTime.now();
    
    // Default constructor for JPA
    public Bookmark() {}
    
    // Constructor with required fields
    public Bookmark(User user, Place place) {
        this.user = user;
        this.place = place;
    }
    
    // Full constructor
    public Bookmark(Long id, User user, Place place, OffsetDateTime createdAt) {
        this.id = id;
        this.user = user;
        this.place = place;
        this.createdAt = createdAt;
    }
    
    // Getters and Setters
    public Long getId() {
        return id;
    }
    
    public void setId(Long id) {
        this.id = id;
    }
    
    public User getUser() {
        return user;
    }
    
    public void setUser(User user) {
        this.user = user;
    }
    
    public Place getPlace() {
        return place;
    }
    
    public void setPlace(Place place) {
        this.place = place;
    }
    
    public OffsetDateTime getCreatedAt() {
        return createdAt;
    }
    
    public void setCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
    }
}

File: src/main/java/com/mohe/spring/entity/Bookmark.kt
package com.mohe.spring.entity

import jakarta.persistence.*
import java.time.OffsetDateTime

@Entity
@Table(name = "bookmarks", uniqueConstraints = [
    UniqueConstraint(columnNames = ["user_id", "place_id"])
])
data class Bookmark(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    val user: User,
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "place_id", nullable = false)
    val place: Place,
    
    @Column(name = "created_at")
    val createdAt: OffsetDateTime = OffsetDateTime.now()
)